articles/model-anatomy.adoc

```diff
@@ -0,0 +1,644 @@
+'SourcePosition <. an2: <<classifier>>
+
+----
+
+An annotation that is applicable to all concepts says it annotates `Node` from LionWeb's built-in library.
+Annotations can _annotate_ other annotations; however, this approach introduces complexities and should be used with great caution.
+
+#TODO:# Do we allow unresolvable annotations?
+
+=== Differences between containment and annotation instance
+Both a contained node `C` and an annotated node `A` share the lifecycle of their parent node `P`.
+So what's the difference?
+
+A containment must be described in the parent's node classifier.
+If the parent is of classifier `House` with containments `rooms`, and `doors`, we cannot add a containment `windows` to the parent node.
+The designer of classifier `House` made the choice that for their domain, a house should only care about rooms and doors, but not windows.
+The designer might not have anticipated our use case of their domain.
+But every user of that classifier relies on this structure of a house, so we must not mess with it.
+
+An annotation instance cannot be mentioned in the parent's node classifier at all.
+Instead, an annotation instance's classifier must be an annotation, which mentions the classifier it annotates.
+Thus, the designer of the annotation decides which targets their annotation is applicable to.
+So we can come up with an annotation `WindowAnnotation` that `annotates: House`, and attach instances to nodes of classifier `House`.
+All these nodes still adhere to the intents of ``House``'s designer, but also transport the additional information about windows we need for our use case.
+
+Even more importantly, we can put orthogonal concerns into annotation instances.
+For the domain of `House`, we don't want to bother with the details how a parser constructed that node.
+But it would be useful to show users where this node originated from in case of errors.
+Not a problem with an annotation `ParserSource` that `annotates: Node` (i.e. any node).
+Similarly, we could come up with an annotation `ReviewComment` to be used during code reviews.
+The reviewer can attach instances to any node, without interfering with the original intent of these nodes.
+
+Review comments demonstrate one drawback of annotations: we're violating the generally valued idea of _separating concerns_, as we mix domain concerns and process concerns in the same model.
+The alternative would be to keep both concerns in separate models, and have the review comments reference the node they apply to.
+Experience shows this approach is very hard to maintain:
+What happens if we deleted a house, and there's still a review comment referring to it?
+The review model might not be loaded during the deletion, so it now contains a "zombie" review comment.
+We might need to introduce "garbage collection" strategies to find such zombies and deal with them.
+For more context-sensitive annotations, moving the annotated node might introduce similar problems.
```

Comment:
Does this paragraph explain why on the M2 nodes contain their annotations?

Reply:
I don't get the question: All annotation instances are contained in their annotated node, both on M1 and M2.

Reply:
I am missing the conclusion of this paragraph: it introduces the challenge/problem but doesn't get to a solution of LionWeb.

Reply:
I came up with an example, does that help?